src/main/java/io/phasetwo/service/resource/UserResource.java

```diff
@@ -59,6 +66,79 @@ public Stream<OrganizationRole> listUserOrgRoles(
     }
   }
 
+  @PUT
+  @Path("/{userId}/orgs/{orgId}/roles")
+  @Consumes(MediaType.APPLICATION_JSON)
+  public Response grantUserOrgRoles(@PathParam("userId") String userId, @PathParam("orgId") String orgId,
+                                    List<OrganizationRole> rolesRep) {
+    log.debugv("Grant user organization roles for %s %s %s", realm.getName(), userId, orgId);
+    UserModel user = session.users().getUserById(realm, userId);
+    OrganizationModel org = orgs.getOrganizationById(realm, orgId);
+    canManage(userId, orgId, user, org);
+    rolesRep.forEach(r -> {
+      OrganizationRoleModel role = org.getRoleByName(r.getName());
+      if (role == null) {
+        throw new NotFoundException(String.format("Organization %s doesn't contain role %s", orgId, r.getName()));
+      }
+      if (!role.hasRole(user)) {
+        role.grantRole(user);
+
+        adminEvent
+                .resource(ORGANIZATION_ROLE_MAPPING.name())
+                .operation(OperationType.CREATE)
+                .resourcePath(session.getContext().getUri())
+                .representation(userId)
+                .success();
+      }
+    });
+    return Response.created(session.getContext().getUri().getAbsolutePathBuilder().build())
+            .build();
+  }
+
+  @DELETE
+  @Path("/{userId}/orgs/{orgId}/roles")
+  @Consumes(MediaType.APPLICATION_JSON)
+  public Response revokeUserOrgRoles(@PathParam("userId") String userId, @PathParam("orgId") String orgId,
+                                     List<OrganizationRole> rolesRep) {
+    log.debugv("Revoke user organization roles for %s %s %s", realm.getName(), userId, orgId);
+    UserModel user = session.users().getUserById(realm, userId);
+    OrganizationModel org = orgs.getOrganizationById(realm, orgId);
+    canManage(userId, orgId, user, org);
+    rolesRep.forEach(r -> {
```

Comment:
As above, need a test that validates when some of the deletes fail:
- everything fails
- no events get emitted
- a proper error gets sent back